client/src/components/AddTrip.tsx

```diff
@@ -0,0 +1,153 @@
+import { useAppDispatch } from '../app/hooks'
+import { Trip } from "../types/Trip";
+import { postTrip } from "../services/apiService";
+import { addTrip } from "../redux/addTripSlice";
+import { ChangeEvent, FormEvent, useState} from "react";
+// import * as dayjs from 'dayjs';
+
+const AddTrip = () => {
+
+    const dispatch = useAppDispatch(); 
+   
+    const [user_id, setUserId] = useState<string>('');
+    const [trip_name, setTripName] = useState<string>('');
+    const [start_loc, setStartLoc] = useState<string>('');
+    const [destination, setDestination] = useState<string>('');
+    const [start_date, setStartDate] = useState<string>('');
+    const [end_date, setEndDate] = useState<string>('');
+
+    const [newTrip, setNewTrip] = useState<Trip>({
+        user_id,  
+        trip_name, 
+        start_loc,  
+        destination, 
+        start_date,
+        end_date,
+    });
+
+    const newTripObj: Trip = {
+        user_id,  
+        trip_name, 
+        start_loc,  
+        destination, 
+        start_date,
+        end_date,
+    }
+
+
+    const handleUserIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setUserId(event.target.value)
+    }
+
+    const handleTripNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setTripName(event.target.value)
+    }
+
+    const handleStartLocChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setStartLoc(event.target.value)
+    }
+
+    const handleDestinationChange = (event: ChangeEvent<HTMLInputElement>) => {

```

Comment:
handleStartLocChange & handleDestinationChange will prolly be have to be changed when inserting maps, hope thats ok!!